src/Libraries/PythonNodeModelsWpf/ScriptEditorWindow.xaml.cs

```diff
@@ -42,6 +42,7 @@ public partial class ScriptEditorWindow : ModelessChildWindow
 
             completionProvider = new SharedCompletionProvider(nodeModel.Engine,dynamoViewModel.Model.PathManager.DynamoCoreDirectory);
             completionProvider.MessageLogged += dynamoViewModel.Model.Logger.Log;
+            nodeModel.CodeMigrated += OnNodeModelCodeMigrated;
```

Comment:
can you unsubscribe this at some point in `nodeView` or `nodeModel's` lifetime? 
I guess it's possible it's not required, but I have to ask for proof - with all the trouble we find with memory leaks during testing.
I like https://blog.jetbrains.com/dotnet/2014/09/04/fighting-common-wpf-memory-leaks-with-dotmemory/ (eventhandler leak section in particular)

Reply:
unsubscribing from this when the ScriptEditorWindow closes now, would that work?

Reply:
I don't think that's a good solution because the script editor window lives for the entire nodeView lifetime I believe. That means it closes multiples times, but is only subscribed once.

Reply:
@mjkkirschner Would you say we should move the event subscription code to the window show event rather than the constructor?

Reply:
that sounds like a good place to put it - or alternatively, clean it up when the nodeView is disposed.

Reply:
I looked at this and it seems like its removing the reference to the `ScriptEditorWindow` when the window is closed and creating a new instance when you double click the node again, or am i missing something?

```
        private void View_MouseDown(object sender, MouseButtonEventArgs e)
        {
            if (e.ClickCount >= 2)
            {
                EditScriptContent(null, null);
                e.Handled = true;
            }
        }

        public void editWindow_Closed(object sender, EventArgs e)
        {
            editWindow = null;
        }

        private void EditScriptContent(object sender, EventArgs e)
        {
            using (var cmd = Dynamo.Logging.Analytics.TrackCommandEvent("PythonEdit"))
            {
                if (editWindow != null)
                {
                    editWindow.Activate();
                }
                else
                {
                    editWindow = new ScriptEditorWindow(dynamoViewModel, pythonNodeModel, pythonNodeView, ref editorWindowRect);
                    editWindow.Initialize(workspaceModel.Guid, pythonNodeModel.GUID, "ScriptContent", pythonNodeModel.Script);
                    editWindow.Closed += editWindow_Closed;
                    editWindow.Show();
                }
            }
        }
```
is there a good way to know when the nodeView is disposed, could only see a way of knowing when the nodeModel is disposed which i guess will also dispose the view?

Reply:
thanks for verifying @SHKnudsen - I think you're right! it looks like the scriptEditor is constructed on edit, if thats the case, I think your change for clearing on window close is good. Let's do that.

Sorry for the confusion.